src/nvim/api/win_config.c

```diff
@@ -240,6 +240,14 @@ void nvim_win_set_config(Window window, Dict(float_config) *config, Error *err)
   // reuse old values, if not overridden
   FloatConfig fconfig = new_float ? FLOAT_CONFIG_INIT : win->w_float_config;
 
+  if (!new_float && HAS_KEY(config, float_config, relative) && striequal(config->relative.data,
```

Comment:
it's a bit weird to replicate already existing parsing logic. I'd do this check on the parsed value in `fconfig` after the call to `!parse_float_config` (or even inside it, passing in win is fine)